experimental/sample.go

```diff
@@ -0,0 +1,109 @@
+package experimental
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+)
+
+//----------------------------------------------------------------------------------
+// PLUGIN HOST
+//----------------------------------------------------------------------------------
+
+// MyHost singelton host service
+type MyHost struct{}
+
+// CheckExeHealth check if the plugin host is running
+func (ds *MyHost) CheckExeHealth(config backend.PluginConfig) *backend.CheckHealthResult {
+	return &backend.CheckHealthResult{
+		Status:  backend.HealthStatusOk,
+		Message: "Plugin is running",
+	}
+}
+
+// NewDataSourceInstance Create a new datasource instance
+func (ds *MyHost) NewDataSourceInstance(config backend.PluginConfig) (DataSourceInstance, error) {
+	settings := myDataSourceSettings{
+		url:  config.DataSourceConfig.URL,
+		port: 1234,
+	}
+
+	return &MyDataSourceInstance{
+		settings: settings,
+		cache:    "TODO",
+		client:   "TODO",
+	}, nil
+}
+
+//----------------------------------------------------------------------------------
+// DATA SOURCE instance
+//----------------------------------------------------------------------------------
+
+type myDataSourceSettings struct {
+	url  string
+	port int32
+}
+
+// MyDataSourceInstance implements backend.DataSourceInstance
+type MyDataSourceInstance struct {
+	settings myDataSourceSettings
+	cache    interface{} // for example
+	client   interface{} // for example
+}
+
+// CheckHealth will check the currently configured settings
+func (ds *MyDataSourceInstance) CheckHealth() *backend.CheckHealthResult {
+	if len(ds.settings.url) < 2 {
+		return &backend.CheckHealthResult{
+			Status:  backend.HealthStatusError,
+			Message: "invalid URL",
+		}
+	}
+	return &backend.CheckHealthResult{
+		Status:  backend.HealthStatusOk,
+		Message: "Datasource is setup properly",
+	}
+}
+
+// QueryData will run a set of queries
+func (ds *MyDataSourceInstance) QueryData(req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	return nil, fmt.Errorf("not implemented")
+}
+
+// CallResource HTTP style resource
+func (ds *MyDataSourceInstance) CallResource(req *backend.CallResourceRequest, sender backend.CallResourceResponseSender) error {
```

Comment:
There's no need to make this a method since `ds` is not in use?
```suggestion
func CallResource(req *backend.CallResourceRequest, sender backend.CallResourceResponseSender) error {
```

Reply:
except that a real instance would... and this is just a sample :)